backend_test.go

```diff
@@ -28,3 +33,175 @@ func getTestBackend(tb testing.TB) (*tfBackend, logical.Storage) {
 
 	return b.(*tfBackend), config.StorageView
 }
+
+var runAcceptanceTests = os.Getenv(envVarRunAccTests) == "1"
+
+type testEnv struct {
+	Token        string
+	Organization string
+	TeamID       string
+	UserID       string
+
+	Backend logical.Backend
+	Context context.Context
+	Storage logical.Storage
+
+	// SecretToken tracks the API token, for checking rotations
+	SecretToken string
+
+	// TokenIDs tracks the IDs of generated tokens, to make sure we clean up
+	TokenIDs []string
+}
+
+func (e *testEnv) AddConfig(t *testing.T) {
+	req := &logical.Request{
+		Operation: logical.CreateOperation,
+		Path:      "config",
+		Storage:   e.Storage,
+		Data: map[string]interface{}{
+			"token": e.Token,
+		},
+	}
+	resp, err := e.Backend.HandleRequest(e.Context, req)
+	assert.Nil(t, resp)
+	assert.Nil(t, err)
+}
+
+func (e *testEnv) AddOrgTokenRole(t *testing.T) {
+	req := &logical.Request{
+		Operation: logical.UpdateOperation,
+		Path:      "role/test-org-token",
+		Storage:   e.Storage,
+		Data: map[string]interface{}{
+			"organization": e.Organization,
+		},
+	}
+	resp, err := e.Backend.HandleRequest(e.Context, req)
+	assert.Nil(t, resp)
+	assert.Nil(t, err)
+}
+
+func (e *testEnv) ReadOrgToken(t *testing.T) {
+	req := &logical.Request{
+		Operation: logical.ReadOperation,
+		Path:      "creds/test-org-token",
+		Storage:   e.Storage,
+	}
+	resp, err := e.Backend.HandleRequest(e.Context, req)
+	assert.NotNil(t, resp)
```

Comment:
Shouldn't these be `require` instead of `assert`? Otherwise the `resp.Data` line below will panic. https://github.com/stretchr/testify#require-package

Reply:
TIL about the `require` package, thanks for the tip. 

I reviewed the tests and determined that `require` is indeed correct, as none of the tests were structured to take advantage of `assert` returning a boolean and continuing. As you noted about the panic above, failing any of those assertions should simply terminate the test. 

Changed in 8d8438b